airflow/utils/db.py

```diff
@@ -561,6 +562,37 @@ def initdb():
     Base.metadata.create_all(settings.engine)  # pylint: disable=no-member
 
 
+def wait_for_migrations(timeout):
+    """
+    Function to wait for all airflow migrations to complete. Used for launching airflow in k8s
+    @param timeout:
+    @return:
+    """
+    from alembic.config import Config
+    from alembic.runtime.migration import MigrationContext
+
+    package_dir = os.path.dirname(importlib.util.find_spec('airflow').origin)
+    directory = os.path.join(package_dir, 'migrations')
+    config = Config(os.path.join(package_dir, 'alembic.ini'))
+    config.set_main_option('script_location', directory)
+    config.set_main_option('sqlalchemy.url', settings.SQL_ALCHEMY_CONN)
+    script_ = ScriptDirectory.from_config(config)
```

Comment:
This shares a lot of code with the "setup" un upgradedb (and also that has some bug fixes that this doesn't.)

We should extract that to a new `_get_alembic_config` function.

Reply:
Fixed